libraries/chain/include/eosio/chain/global_property_object.hpp

```diff
@@ -153,16 +153,23 @@ namespace eosio { namespace chain {
 
    inline snapshot_global_property_object::extension_t get_gpo_extension(const global_property_object& gpo) {
       return snapshot_global_property_object::extension_v0{
-          gpo.proposed_security_group_block_num, gpo.proposed_security_group_participants, gpo.transaction_hooks};
+          gpo.proposed_security_group_block_num,
+          {gpo.proposed_security_group_participants.begin(), gpo.proposed_security_group_participants.end()},
+          {gpo.transaction_hooks.begin(), gpo.transaction_hooks.end()}};
    }
 
    inline void set_gpo_extension(global_property_object&                        gpo,
                                  snapshot_global_property_object::extension_t&& extension) {
       std::visit(
           [&gpo](auto& ext) {
              gpo.proposed_security_group_block_num    = ext.proposed_security_group_block_num;
-             gpo.proposed_security_group_participants = std::move(ext.proposed_security_group_participants);
-             gpo.transaction_hooks                    = std::move(ext.transaction_hooks);
+             gpo.proposed_security_group_participants = {ext.proposed_security_group_participants.begin(),
+                                                         ext.proposed_security_group_participants.end(),
+                                                         gpo.proposed_security_group_participants.key_comp(),
+                                                         gpo.proposed_security_group_participants.get_allocator()};
+             gpo.transaction_hooks                    = {ext.transaction_hooks.begin(),
+                                                         ext.transaction_hooks.end(),
+                                                         gpo.transaction_hooks.get_allocator()};
```

Comment:
same here as for `proposed_security_group_participants ` comment

Reply:
See other comment.

Reply:
got it. if you don't need allocator and comparator to be custom instead of instantiating it from template parameter by default -  please remove.

Reply:
Are you saying that there is a constructor that allows you to default the comparator, while setting the allocator and providing iterators? If there is point me to it. Allocator has to come from the original container in the global_property_object.
If you are wondering why transaction_hook doesn't worry about comparator, that is because vector doesn't have a comparator.